test/type_prop.cpp

```diff
@@ -2385,6 +2386,44 @@ TEST(type_prop, or_bad_arguments)
         });
 }
 
+TEST(type_prop, embedding_lookup_static_shapes)
```

Comment:
These tests are good but we need to implement some tests for failing cases as well.